### ISSUE TITLE
Change remote features transient to a persistent option

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#4861]
Following #4856 and #4858, now that the `/site-management/features/` request is invoked regularly in a cron job, it is no longer necessary to use a transient, and it can become a (persistent) option instead. The main advantage of that is that the data about enabled features never expires, but is still regularly updated. This ensures that e.g. enabled features remain enabled even when the site is no longer connected (e.g. after a reset), which is critical for certain features, e.g. the `serviceSetupV2` (see #4860).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `googlesitekit_remote_features` transient should be changed to a `googlesitekitpersistent_remote_features` _option_.
    * Including the `googlesitekitpersistent_` prefix is crucial here so that this remains in place even after a reset.
* The temporarily added option `googlesitekitpersistent_service_setup_v2_enabled` should no longer be set or modified, since the above change of using a persistent option for the overall remote features resolves this problem holistically now. **However**, a check for its value still needs to remain in place so that the transition to the new option is seamless.

## Implementation Brief

Within `includes/Core/Authentication/Authentication.php`:

* The dependency #4856 mentions extracting the logic to set the transient to a new method, called, e.g. `get_transient_features`. Locate this method. If the name does include the word _transient_, rename it to something more appropriate - for example, `get_persistent_features`.
* Within this method:
  * Remove the check for `false === $features`. The features should be persisted every time, as they no longer expire.
  * Replace the use of `$this->transients` with `$this->options`, removing the expiration argument.
  * The option name should be `googlesitekitpersistent_remote_features`.
  * If an error is returned from the call to `get_features`, don't set the option, but rather delete it. That way, the features will be checked again next time this method is called on-demand, rather than having to wait for half a day with a stale/empty feature set.
  * Remove the code to set the `googlesitekitpersistent_service_setup_v2_enabled` option.
* In the `filter_features_via_proxy` method:
  * Ensure the `googlesitekitpersistent_remote_features` option is checked within this method, and only if it returns `false` (i.e. not set), then invoke `get_persistent_features` to retrieve the features on-demand.

### Test Coverage

* No new tests needed.

### QA
* Set the `Force enabled feature flags` to 'Do not override' in the Google SK Tester Plugin. In a fresh WP install, activate SK. Verify that the Unified Dashboard is enabled.
___

### QA: Eng
* Configure HTTP proxy on your local machine following instructions in the [engineering workflow doc](https://docs.google.com/document/d/1vYRA-YxxZdufO6piF0gA1k60N3UxbvuqyQr_XfGgvSk/edit#heading=h.faf0umy03arh) to use mitmproxy or use an equivalent tool to inspect outgoing HTTP requests.
* Create a fresh install of WP (or Reset Site Kit and clear all site data in the browser) and install + activate the SK plugin. 
    - Verify that no requests were made to the `/site-management/features/` endpoint. 
    - Verify that [the DB has a new cron job](https://wordpress.stackexchange.com/questions/98032/is-there-a-quick-way-to-view-the-wp-cron-schedule) - `googlesitekit_cron_update_remote_features`. 
    - Verify the old `get_transient_features` cron hook is not added.
    - Verify that the old `_transient_googlesitekit_remote_features` option is not present in the `wp_options` table.
    - Verify that the new `googlesitekitpersistent_remote_features` option is not present in the `wp_options` table.
* Now setup site kit. While doing so: 
    - Verify that a request is made to `/site-management/features/` and the response contains the enabled features.
    - Once setup is complete, verify that the `googlesitekitpersistent_remote_features` option is present in the `wp_options` table and contains the enabled features.

## Changelog entry

- Store remote features as a persistent option, ensuring features remain enabled when Site Kit is disconnected or reset.

**Comment by tofumatt** [comment]
Looks good to me.

IB ✅

**Comment by jimmymadon** [comment]
I will have to slightly update the PR to this issue once #4957 is merged into main. Otherwise this is good to go.

**Comment by hussain-t** [comment]
## QA Verified ✅ 

- When the plugin is installed and activated before the setup, verified the following:
    - No requests were made to the `/site-management/features/` endpoint.
    - The DB has a new cron job - `googlesitekit_cron_update_remote_features`.
    - The old `get_transient_features` cron hook is not added.
    - The old `_transient_googlesitekit_remote_features` option is not present in the `wp_options` table.
    - The new `googlesitekitpersistent_remote_features` option is not present in the `wp_options` table.
- After the setup verified the following:
    - A request is made to `/site-management/features/` and the response contains the enabled features.
    - The `googlesitekitpersistent_remote_features` option is present in the `wp_options` table and contains the enabled features.
 
<img width="1775" alt="Screenshot 2022-03-24 at 3 45 41 PM" src="https://user-images.githubusercontent.com/24408261/159896552-774b6ddd-391b-4401-96ee-c0ab7abe2430.png">

**Comment by felixarntz** [comment]
### Approval ❌ 

This may be a misunderstand from the ACs, but the second point of the ACs states that the workaround with that `serviceSetupV2`-specific option (which is now removed anyway due to #4950) is no longer needed _because_ the option is persistent. However, the implementation in #4964 [deletes the option in case of an error](https://github.com/google/site-kit-wp/pull/4964/files#diff-fd8b5c88edabac5e94ebf3751d5ea639f149dec14a1296a74b6890617a92bf2dR1438), which breaks that promise. One purpose of the option is that it remains present even when the site is disconnected or when a temporary error occurs. So in case of an error, it should remain untouched so that the user at least keeps the previously active features.

**Comment by jimmymadon** [comment]
Ahh good point - should've caught this. I thought the point in the IB made sense and just followed along. I have created a follow up PR to fix this.

**Comment by felixarntz** [comment]
@jimmymadon Thanks for the quick fix! All good branching of `main`: Since this is a follow-up fix for a release issue, it actually _has_ to go into the release :)

**Comment by felixarntz** [comment]
### Approval ✅
